cloudify_agent/app.py

```diff
@@ -124,3 +137,8 @@ def new_excepthook(exception_type, value, the_traceback):
         current_excepthook(exception_type, value, the_traceback)
 
     sys.excepthook = new_excepthook
+    utils.enable_cluster_failover(app)
```

Comment:
always enable failover and register the `update` task - if the manager isn't a cluster, the failover strategy is equivalent to round-robin (and anyway, it'll just have one broker address), and the task will never be called.

This also means an agent can be reconfigured at runtime - if it was installed with just one manager, it will still be able to switch to a cluster later.